Add test program

aidanheerdegen opened the pull request:
This PR adds a small FORTRAN test program to this repo as part of addressing #26.

Tested by building
```
cmake -B build
cmake --build build
```

and running
```
$ ./build/src/stub.exe 
 Running ACCESS-TEST stub
$
```

One question: is it a problem having a `CMakeLists.txt` in the top level? Should this and the source code all be in a subdirectory, e.g. `test` or `stub`?

aidanheerdegen committed:
Add FORTRAN stub program and CMake build infra

aidanheerdegen committed:
Fix errors in exe path

aidanheerdegen committed:
Trim debug string output

aidanheerdegen commented:
Most of my CMake knowledge has evaporated out of my brain, so I followed the instructions here for making a simple CMake FORTRAN project

https://www.atomwitch.net/2022/04/02/cmake-fortran.html

CodeGat reviewed (commented):
Looks fine to me, although I'm no fortran expert :laughing: What is a CMakeLists.txt file? Is it similar to a Makefile?

aidanheerdegen committed:
Move stub source and make infra into sub-directory

aidanheerdegen committed:
Replaced code with hello world MPI code

aidanheerdegen committed:
Moved source code name to reflect change to hello world code

aidanheerdegen committed:
Update cmake to use a single file and udpated to use new source code filename

aidanheerdegen committed:
Add MPI support to build and source code

aidanheerdegen commented:
Ok, I've modified the code to use a simple hello world MPI implementation. I've also moved the code and build infra into a `stub` subdirectory.

On`gadi` the following is sufficient to compile:
```
module load openmpi
FC=mpif90 ccmake -B build
FC=mpif90 cmake --build build
```
Sample run:
```
$ mpirun -n 6 build/hello_world.exe 
 Hello World from process:            0 of            6
 Hello World from process:            1 of            6
 Hello World from process:            2 of            6
 Hello World from process:            3 of            6
 Hello World from process:            4 of            6
 Hello World from process:            5 of            6
$
```
(NB: Also produced some noisy complaints about `shmem: mmap` and `create_and_attach: unable to create shared memory BTL coordinating structure :: size 134217728` but it didn't stop it running)
